services/QuillLMS/app/models/unit.rb

```diff
@@ -47,6 +48,8 @@ class Unit < ApplicationRecord
 
   default_scope { where(visible: true)}
 
+  before_save :set_active_to_false
```

Comment:
I'm a little torn on this one.  It feels more Rails-y to handle this with a validator function that marks the model as invalid if `visible == false && active == true`, which will prevent the model from saving if that illegal combination is set.  However, the nice thing about your approach is that any code that sets `visible == false`, it automatically "fixes" any values for `active` that aren't also set at the same time.

Basically, I'm not sure if the "magic" of doing this in a `before_save` call is better because it lets us fix most potential cases of conflict in one place, but it's also "magic", and could theoretically surprise someone.  That said, the "surprise" being something like "you can establish an invalid data state" probably isn't a bad one?

I'd be interested to hear other peoples' thoughts on this one.

Reply:
Given our reliance on setting the visible flag in other callbacks, it's probably easiest to use a callback here.